src/mlpack/bindings/cli/add_to_po.hpp

```diff
@@ -14,95 +13,171 @@
+      },
+      param.desc.c_str());
 }
 
 /**
- * Add a boolean option to boost::program_options.
+ * Add a boolean option to CLI11.
  *
- * @param boostName The name of the option to add to boost::program_options.
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
-             const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
-             const typename boost::enable_if<std::is_same<T, bool>>::type* = 0)
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
+             const typename boost::enable_if<std::is_same<T, bool>>::type* = 0,
+             const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0,
+             const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+             const typename boost::disable_if<std::is_same<T,
+                std::tuple<mlpack::data::DatasetInfo, arma::mat>>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(), descr.c_str());
+  app.add_flag_function(cliName.c_str(),
+      [&param](const T& value)
+      {
+        param.value = value;
+        param.wasPassed = true;
+      },
+      param.desc.c_str());
 }
 
 /**
- * Add an option to boost::program_options.  This is the function meant to be
+ * Add an option to CLI11.  This is the function meant to be
  * used in the CLI function map.
  *
  * @param d Parameter data.
  * @param * (input) Unused void pointer.
  * @param output Void pointer to options_description object.
  */
 template<typename T>
-void AddToPO(const util::ParamData& d,
+void AddToPO(util::ParamData& param,
              const void* /* input */,
              void* output)
 {
-  // Cast boost::program_options::options_description object.
-  boost::program_options::options_description* desc =
-      (boost::program_options::options_description*) output;
+  // Cast CLI::App object.
+  CLI::App* app = (CLI::App*) output;
 
-  // Generate the name to be given to boost::program_options.
+  // Generate the name to be given to CLI11.
   const std::string mappedName =
-      MapParameterName<typename std::remove_pointer<T>::type>(d.name);
-  std::string boostName = (d.alias != '\0') ? mappedName + "," +
-      std::string(1, d.alias) : mappedName;
+      MapParameterName<typename std::remove_pointer<T>::type>(param.name);
+  std::string cliName = (param.alias != '\0') ?
+      "-" + std::string(1, param.alias) + ",--" + mappedName : "--" + mappedName;
 
   // Note that we have to add the option as type equal to the mapped type, not
-  // the true type of the option.
+  // the true type of the option.  
   AddToPO<typename ParameterType<typename std::remove_pointer<T>::type>::type>(
```

Comment:
`typename ParameterType<typename std::remove_pointer<T>::type>::type` is not the same as the actual type of the option, so the SFINAE checks in the other implementations don't work.  We would just use `T` here, but, actually, for serializable models, we get `T` as a pointer, and SFINAE is easier to do without pointers, so use `typename std::remove_pointer<T>::type` here, as is done in `set_param.hpp`. :+1: